lib/elixir/src/elixir_errors.erl

```diff
@@ -65,7 +65,7 @@ emit_diagnostic(Severity, Position, File, Message, Stacktrace) ->
 
 fancy_exception_snippet(File, LineNumber, Column, Description, Snippet) ->
   #{content := Content, offset := Offset} = Snippet,
-
+  {TrimmedContent, TotalLeading} = trim_line(Content),
```

Comment:
Do you think it is worth trimming? It may give a wrong impression on how far the content actually is, no?

Reply:
I'll leave that up to you, I decided to trim it because I think having the snippet and a pointer on the error token is already enough for me to locate it fast, and because I was also trimming the lines read from the files on the diagnostic emitting code. I like that the errors look more compact and solid when the error is always at the left.

For comparison:
```
** (SyntaxError) invalid syntax found on nofile:5:11:
   ┌─ error: nofile:5:11
   │
 5 │ end
   │ ^
   │
   unexpected reserved word: end
```

```
** (SyntaxError) invalid syntax found on nofile:5:11:
   ┌─ error: nofile:5:11
   │
 5 │                                      end
   │                                      ^
   │
   unexpected reserved word: end
```

Reply:
Can we check if Elm/Rust trim it or not?

Reply:
None of them trims it, so I think we can follow it. I'll revert the trimming commit.

Elm:
![image](https://github.com/elixir-lang/elixir/assets/59743220/f7d5be75-fb9c-4596-bd71-e0db2d3b75f8)

Rust:
![image](https://github.com/elixir-lang/elixir/assets/59743220/be9f9efc-01b8-4753-b0b1-96cccc4c3541)

Rust adds a `...` at the start of the line to better represent the empty space I guess, what do you think of it?

Reply:
I like Rust's route. But I assume they have a threshold. How many spaces you need to have before they starting trimming?

Reply:
It starts trimming the line to `...<19 spaces><line>` when there are >= 27 leading spaces on it

Reply:
Sounds good to me!